kani-compiler/src/codegen_cprover_gotoc/codegen/function.rs

```diff
@@ -123,6 +123,7 @@ impl<'tcx> GotocCtx<'tcx> {
             self.symbol_table.update_fn_declaration_with_definition(&name, body);
 
             self.handle_kanitool_attributes();
+            self.record_test_metadata();
```

Comment:
Is "test" the right word here? It's hard to tell from this function name what its doing: my first thought was that it was recording some metadata related to testing kani.

Reply:
Also, does this really belong as part of codegen? It might make more sense to do this either in its own pass, or when we declare the function, rather than as part of codegenning it.

Reply:
Perhaps not? But this is also where we spot proof harnesses, so these are in the same piece of code here.